soufi/finders/ubuntu.py

```diff
@@ -70,7 +68,7 @@ def populate_archive(self, temp_dir, tar):
         names = [url.rsplit('/', 1)[-1] for url in self.urls]
         for name, url in zip(names, self.urls):
             arcfile_name = self.download_file(
-                temp_dir, name, url, timeout=API_TIMEOUT
+                temp_dir, name, url, timeout=UbuntuFinder.TIMEOUT
```

Comment:
Ok so the changes in this file have solidified a spark of a thought I was having from my earlier comments about the TIMEOUT value.
Instead of using a class property in the base class, let's make it an instance property that's set in `__init__` with a default of 30, and then just refer to self.timeout everywhere.

Reply:
I considered it, and it means passing the value from `SourceFinder`-derived objects into `DiscoverdSource`-derived objects, which means fidgeting with method signatures everywhere.

I figured class properties would look cleaner, but looking back at it, it's all pretty gnarly.  An instance property would also facilitate overriding it, which might be handy.

Reply:
Yeah overriding was my main thought. It should be easy enough to pass the same value into the `DiscoveredSource`.